### PR TITLE
Various Fixes

### PR DESCRIPTION
**Pull request by pshriwise**
I ran across a few things recently and grouped them up into this PR.

  - The `openmc-plot-mesh-tally` script was broken for 2D meshes. The set of indices used to represent 2D meshes is size 2. There is a fix in there to handle that.

  - A coefficient lookup in one of the examples is no longer valid now that surfaces use `r` rather than `R`.

  - A couple of jupyter examples use indexes that are no longer valid for some of our data structures in statepoint files.

  - This last one was a deeper and I'm not totally sure of my fix TBH, so if someone can verify it that would be grand. The `mdgxs-part-i.ipynb` example resulted in a seg-fault on `openmc.run()`. This turned out to be caused by an invalid look up of a non-existant delayed group filter coming from an `openmc.mdgxs.ChiDelayed` filter in the example. In `score_fission_eout`, `tally.delayedgroup_filter_` is -1 but `i_dg_filt` is set to 0 by default for an out-of-range index on the tally's filter vector. I've updated that check to look at `tally.delayedgroup_filter_` instead.

### CODE REVIEW

File: examples/jupyter/search.ipynb [hunk 1]
```diff
@@ -139,24 +135,160 @@
...
-    "collapsed": false
-   },
+   "metadata": {},
    "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/shriwise/.pyenv/versions/3.6.7/lib/python3.6/site-packages/openmc-0.10.0-py3.6-linux-x86_64.egg/openmc/surface.py:1300: FutureWarning: \"ZCylinder(...) accepts an argument named 'r', not 'R'. Future versions of OpenMC will not accept the capitalized version.\n",
+      "  warn(_WARNING_UPPER.format(type(self).__name__, 'r', 'R'), FutureWarning)\n",
```
Comment (paulromano) [inline examples/jupyter/search.ipynb]: Can you fix this warning while we're at it?

File: examples/python/pincell_depletion/run_depletion.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 time_step = 1*24*60*60 # s
 final_time = 5*24*60*60 # s
 time_steps = np.full(final_time // time_step, time_step)
-
+print(time_steps)
```
Comment (paulromano) [inline examples/python/pincell_depletion/run_depletion.py]: Remove

File: scripts/openmc-plot-mesh-tally [hunk 3]
```diff
@@ -216,24 +216,42 @@ class MeshPlotter(tk.Frame):
...
+            h_ind = 1
+            v_ind = 2
         else:
-            dims = (self.nx, self.nz)
+            h_ind = 0
+            v_ind = 2
+
+        axial_ind = 3 - (h_ind + v_ind)
+        print(h_ind, v_ind)
+        print(dims)
```
Comment (paulromano) [inline scripts/openmc-plot-mesh-tally]: Remove

File: src/tallies/tally_scoring.cpp [hunk 4]
```diff
@@ -257,14 +257,16 @@ score_fission_eout(const Particle* p, int i_tally, int i_score, int score_bin)
       #pragma omp atomic
       tally.results_(filter_index, i_score, RESULT_VALUE) += score;
 
-    } else if (score_bin == SCORE_DELAYED_NU_FISSION && g != 0) {
-
-      // Get the index of the delayed group filter
-      auto i_dg_filt = tally.filters()[tally.delayedgroup_filter_];
+    } else if (score_bin == SCORE_DELAYED_NU_FISSION &&
+               g != 0) {
```
Comment (paulromano) [inline src/tallies/tally_scoring.cpp]: I'd keep this on the same line as above

File: examples/jupyter/search.ipynb (hunk 1 above)
Comment (pshriwise) [reply examples/jupyter/search.ipynb]: Can do!

File: examples/python/pincell_depletion/run_depletion.py (hunk 2 above)
Comment (pshriwise) [reply examples/python/pincell_depletion/run_depletion.py]: Left a few hasty debug statements in here... sorry!

**Comment by paulromano** [comment]
Looks good. I'll wait for the 36 hour review period to finish before merging.

**Pull request merged**